### PR TITLE
Fix branch coverage tracking and add coverage testing

### PR DESCRIPTION
**Pull request by benber86**
### What I did

Branch coverage barely worked with conditional statements/loops/multiline statements as described in #327.

### How I did it

- Replaced ast-based coverage trace approximation with bytecode based one
- Used fuzzer + mutation test based harness to get claude to find, fix and write tests for all edge-cases

### How to verify it

`python -m pytest tests/unitary/test_coverage`

### Description for the changelog

Fix branch coverage and expand test suite

### Cute Animal Picture
image

### CODE REVIEW

File: boa/vm/py_evm.py [hunk 1]
```diff
@@ -410,6 +429,7 @@ def _init_vm(self, account_db_class=AccountDB):
         # patch in tracing opcodes
         c.opcodes[0x20] = Sha3PreimageTracer(c.opcodes[0x20], self.env)
         c.opcodes[0x55] = SstoreTracer(c.opcodes[0x55], self.env)
+        c.opcodes[0x57] = JumpiTracer(c.opcodes[0x57])
```
[the remark this answers is not in the document]
Comment (charles-cooper) [reply boa/vm/py_evm.py]: it can be wrapped multiple times though
